kayak_core/src/layout_dispatcher.rs

```diff
@@ -0,0 +1,65 @@
+use indexmap::IndexSet;
+use morphorm::{Cache, GeometryChanged};
+
+use crate::{Index, KayakContext, KayakContextRef, LayoutEvent};
+
+pub(crate) struct LayoutEventDispatcher;
+
+impl LayoutEventDispatcher {
+    pub fn dispatch(context: &mut KayakContext) {
+        let dirty_nodes: Vec<Index> = context
+            .widget_manager
+            .dirty_render_nodes
+            .drain(..)
+            .collect();
+
+        // Use IndexSet to prevent duplicates and maintain speed
+        let mut parents: IndexSet<Index> = IndexSet::default();
+
+        for node_index in dirty_nodes {
+            // If layout is not changed -> skip
+            if context
+                .widget_manager
+                .layout_cache
+                .geometry_changed(node_index)
+                == GeometryChanged::default()
```

Comment:
You _might_ be able to do this?

```suggestion
            if context
                .widget_manager
                .layout_cache
                .geometry_changed(node_index)
                .is_empty()
```

Alternatively, this could also be done as a `.filter(...)` after the `.drain(...)`.

Not super important either way haha, so whichever you think is clearest.